File: PcmBackend/Models/RankHistory.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PcmBackend.Models;

[Table("643_RankHistory")]
public class RankHistory
{
    [Key]
    public int Id { get; set; }
    
    public int MemberId { get; set; }
    [ForeignKey("MemberId")]
    public Member Member { get; set; }

    public double RankLevel { get; set; }
    
    public string Reason { get; set; } = string.Empty; // e.g., "Match result", "Admin adjustment"
    
    public DateTime CreatedDate { get; set; } = DateTime.Now;
}

File: pcm_mobile_643/lib/models/rank_history.dart
class RankHistory {
  final int id;
  final int memberId;
  final double rankLevel;
  final String reason;
  final DateTime createdDate;

  RankHistory({
    required this.id,
    required this.memberId,
    required this.rankLevel,
    required this.createdDate,
    required this.reason,
  });

  factory RankHistory.fromJson(Map<String, dynamic> json) {
    return RankHistory(
      id: json['id'],
      memberId: json['memberId'],
      rankLevel: (json['rankLevel'] ?? 0).toDouble(),
      reason: json['reason'] ?? '',
      createdDate: DateTime.parse(json['createdDate']),
    );
  }
}
